### PR TITLE
p3: Add `SourceChip`s components

### PR DESCRIPTION
**Pull request by ab-gnm**
#### Description

**Figma**: https://www.figma.com/design/kSmjgtoTWiG8N7HXxFoGEE/%E2%97%90-Apps-library?node-id=6847-12224&node-type=section&t=P7PdEYK5Rj2d9YYm-11

This PR adds the `SourceChip` component.

**Key changes**

1. Added new `SourceChip` and `SourceMultiSelectChip` components.
   * Chips support two predefined sizes - `Small` and `Medium`
   * Default styling matches source documentation. Colours and borders can be overridden by developers
   * Default badge matches source documentation. A variant allows developers to provide a custom badge.
   * Default variants only allow an icon before the label. Advanced variants allow a custom after label icon.
2. Added `HorizontalExpandingText` component that animates text change by animating expansion/collapse horizontally.
3. Added `ChipIndicator` sealed type to indicate whether we're displaying an `Icon`, an `Image` or `None`. Icons and images get different height and padding. Using a sealed type simplifies applying correct styles in code.
   * Icons and Images get three sub types - `Vector`, `Painter` and `Component`. This makes it easy to reuse these with either an `ImageVector`, a `Painter` or a custom component without overriding the based functions.
4. Added `SourceBaseChip`. It only applies the `style`. Everything else can be overridden. It only takes a single `content` lambda instead of text, and before/after indicators.

Some other changes:

1. Moved preview components in `:sample` inside `previews` package
2. Added `lint` checks in CI
3. Copied over `DensityExtension`s from news app repo

#### Testing notes/instructions:

Use the chips preview in sample app to confirm chips work as expected. 

#### Checklist
- [X] Changes have been checked by the developer
- [ ] Changes have been checked by the reviewers
- [X] Unit tested

#### For pull requests introducing UI changes:

- [ ] Sign-off by Design: @akemitakagi 
- [X] Dark Mode 
- [X] Tablet 
- [X] Accessibility (e.g. VoiceOver): 

##### Screenshots or videos:

| `Light` | `Dark` |
| --- | --- |
|light|dark|

##### Spacing

Two assumptions not from markup:
1. horizontal padding for label only chips is 16.dp on both sizes
2. horizontal padding for icon only chips is 12.dp on both sizes

**Start**

| Spacing | Size | With Icon | With Image | Label only | Only Icon |
|---|---|---|---|---|---|
| Before icon/image | Small | 8 | 8 | 16 | 12 |
| Before icon/image | Medium | 12 | 12 | 16 | 12 |
| Between icon/image and text | Small | 4 | 8 | 0 | 0 |
| Between icon/image and text | Medium | 4 | 8 | 0 | 0 |

**End**

| Spacing | Size | With check icon | Label only | Only Icon |
|---|---|---|---|---|
| Between text and check icon | Small | 8 | 0 | 0 |
| Between text and check icon | Medium | 8 | 0 | 0 |
| After check icon | Small | 8 | 16 | 12 |
| After check icon | Medium | 8 | 16 | 12 |

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -49,8 +49,7 @@ jobs:
       - name: Run ktlint
         run: |
           cd android
-          ./gradlew lintKotlin --continue
-          ./gradlew :build-logic:convention:lint --continue
+          ./gradlew lint lintKotlin :build-logic:convention:lint --continue
```
Comment (ab-gnm) [inline .github/workflows/lint.yml:52]: Combined both to reduce gradle load time. Also added `lint` check on library source.

File: android/gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [versions]
 group = "com.gu.source"
-libraryVersion = "0.5.0"
+libraryVersion = "0.6.0"
```
Comment (ab-gnm) [inline android/gradle/libs.versions.toml:3]: Updated minor version to reflect addition of chips components

File: android/sample/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -17,7 +17,6 @@
         <activity
             android:name="com.gu.source.MainActivity"
             android:exported="true"
-            android:label="@string/app_name"
```
Comment (ab-gnm) [inline android/sample/src/main/AndroidManifest.xml:20]: Outcome of lint - removed duplicate label (same string exists on `application`)

File: android/source/src/main/kotlin/com/gu/source/components/HorizontalExpandingText.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @param text The text to display.
+ * @param colour The color of the text.
+ * @param style The [TextStyle] to apply to the text.
+ * @param modifier A [Modifier] to apply to the [Text] composable.
+ * @param maxLines Maximum number of lines for the text.
+ * @param overflow Defines how the text should handle overflow when the text is too long.
+ */
+@Composable
+fun HorizontalExpandingText(
```
Comment (ab-gnm) [inline android/source/src/main/kotlin/com/gu/source/components/HorizontalExpandingText.kt:28]: Used to provide expand collapse animation on a chip when text is removed. Used for the "All" chip on MyG v2 in the news app.

**Comment by akemitakagi** [comment]
Hi @ab-gnm , 
Some minor padding adjustments are needed for the icon and image chips. 
Please check this mark-up.
Let me know if you have any questions!

File: android/source/src/main/kotlin/com/gu/source/components/chips/ChipDecoration.kt [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+ * tinting, use [ChipDecoration.Icon.Component] or [ChipDecoration.Image.Component] and provide a
+ * custom composable function with the desired colour.
+ *
+ * For icons, prefer to use [ChipDecoration.Icon.Vector] with a [Source.Icons] vector where possible.
+ * Alternatively use [ChipDecoration.Icon.Painter] with a drawable resource ID.
+ *
+ * For images, prefer to use [ChipDecoration.Image.Painter] for static images. For images fetched
+ * from the network, use [ChipDecoration.Image.Component] with a custom composable function.
+ */
+sealed class ChipDecoration {
```
Comment (lungc) [inline android/source/src/main/kotlin/com/gu/source/components/chips/ChipDecoration.kt:40]: Great improvement, this make the chip much more flexibility.maybe we can add  alignment options  and shape later.

**Review by lungc — APPROVED** [review]
Great improvement and refactor

**Pull request merged**